cf-agent/package_module.c

```diff
@@ -742,14 +742,17 @@ int UpdatePackagesDB(Rlist *data, const char *pm_name, UpdateType type)
         char *inventory_list = BufferClose(inventory_data);
 
         /* We can have empty list of installed software or available updates. */
-        if (!inventory_list)
+        if (inventory_list == NULL)
         {
-            WriteDB(db_cached, inventory_key, "\n", 1);
+            WriteDB(db_cached, inventory_key, "", 0);
         }
         else
         {
             WriteDB(db_cached, inventory_key, inventory_list,
                     strlen(inventory_list));
+        }
+        if (inventory_list != NULL)
+        {
             free(inventory_list);
```

Comment:
This change was not necessary, you could leave it inside the `else`. Also, as it is now, you could just drop the if. Calling `free(NULL)` is okay.

Reply:
Right. That was another hold-over from a previous version where inventory_list could be != NULL at that point.